src/main/java/org/apache/sling/api/resource/mapping/PathToUriMappingService.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.sling.api.resource.mapping;
+
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.apache.sling.api.uri.SlingUri;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.osgi.annotation.versioning.ProviderType;
+
+/**
+ * Provides a way to resolve and map paths to Sling URIs. Both operations are extensible by
+ * {@link org.apache.sling.spi.urimapping.SlingUriMapper} OSGi services.
+ * 
+ * @since 1.1.0 (Sling API Bundle 2.23.0)
+ */
+@ProviderType
+public interface PathToUriMappingService {
+
+    /** The result of a map or resolve operation */
+    @ProviderType
+    public interface Result {
+        /**
+         * The Sling URI as result of the resolve or map operation.
+         * 
+         * @return the Sling URI
+         */
+        @NotNull
+        SlingUri getUri();
+
+        /**
+         * Returns all intermediate mappings as produced by {@link org.apache.sling.spi.urimapping.SlingUriMapper} services
+         * 
+         * @return the intermediate mappings
+         */
+        @NotNull
+        Map<String, SlingUri> getIntermediateMappings();
```

Comment:
Why is this a map and not a list? What is the key for the values?

Reply:
So currently the key is the classname of the `SlingUriMapper` that produced the result, it would probably be better to use type `Class<? extends SlingUriMapper>` as key, that would make it automatically clearer. 

The map is immutable as the constructor call 
https://github.com/apache/sling-org-apache-sling-resourceresolver/blob/4ab43acf49dfd9cd25119b74054980b3c4144918/src/main/java/org/apache/sling/resourceresolver/impl/urimapping/PathToUriMappingServiceImpl.java#L108 
uses
 https://github.com/apache/sling-org-apache-sling-resourceresolver/blob/4ab43acf49dfd9cd25119b74054980b3c4144918/src/main/java/org/apache/sling/resourceresolver/impl/urimapping/MappingChainContextInternal.java#L54

Reply:
I don't think that a class name is a good key - in theory there could be a factory component for SlingUriMapper with different configurations - sharing the same class name.
As this is ordered, it probably should be a list. So how about a list of some type of object, which has two getter methods, one to get the SlingUri, the other to get some info about the SlingUriMapper. We could add a default method to SlingUriMapper like getName() which by default returns the class name. But this way, the contract is at least more clear.

Reply:
Updated the `getIntermediateMappings()` method in 43f580496978061a3dcd92962928a257ee2ca311 (for now left out `SlingUriMapper.getName()`, I think this can be autogenerated and if needed, be overwritten by a OSGI service property like we do it for HCs, that's maybe cleaner than forcing everybody to implement `getName()`)